xbmc/platform/linux/SysfsPath.cpp

```diff
@@ -0,0 +1,29 @@
+/*
+ *  Copyright (C) 2011-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "SysfsPath.h"
+
+#include <unistd.h>
+
+namespace
+{
+const int PAGESIZE = getpagesize();
+}
+
+template<>
+bool CSysfsPath::Set(std::string value)
```

Comment:
Are you sure this function gets called when you don't declare it in the header?

Reply:
Yes I am, The tests prove this and so does setting a breakpoint with gdb

```
(gdb) break CSysfsPath::Set<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) 
Breakpoint 1 at 0x19cc03d: CSysfsPath::Set<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >). (2 locations)
(gdb) r
Starting program: /home/lukas/Documents/git/xbmc/build/kodi-test --gtest_filter=TestSysfsPath\*

[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
Missing separate debuginfo for /lib64/libwayland-cursor.so.0

Note: Google Test filter = TestSysfsPath*
[==========] Running 2 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 2 tests from TestSysfsPath
[ RUN      ] TestSysfsPath.String

Breakpoint 1, CSysfsPath::Set<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > (this=0x5a0a090, value='0' <repeats 200 times>...) at /home/lukas/Documents/git/xbmc/xbmc/platform/linux/SysfsPath.cpp:34
warning: Source file is more recent than executable.
34	  if (static_cast<int>(value.size()) > PAGESIZE)
```

Reply:
Did you try it in the release build?

Reply:
This function doesn't get called anywhere else then the test.